frontend/src/components/ShootWorkers/WorkerInputGeneric.vue

```diff
@@ -264,14 +272,43 @@ export default {
     volumeInCloudProfile () {
       return !isEmpty(this.volumeTypes)
     },
+    selectedMachineType () {
+      return find(this.machineTypes, ['name', this.worker.machine.type])
+    },
+    canDefineVolumeSize () {
+      // Volume size can be configured by the user if the volume type is defined via a volume type (volumeInCloudProfile)
+      // not via machine type storage. If defined via storage with type not 'fixed' or if no storage is present, then the
+      // user is allowed to set a volume size, in other words storage.machine.type must not be 'fixed'
+      if (this.volumeInCloudProfile) {
+        return true
+      }
+      const storage = get(this.selectedMachineType, 'storage')
+      if (!storage.type) {
+        return true
+      }
+      if (storage.type !== 'fixed') {
+        return true
+      }
+      return false
+    },
     machineImages () {
       return filter(this.machineImagesByCloudProfileName(this.cloudProfileName), ({ isExpired }) => {
         return !isExpired
       })
     },
     minimumVolumeSize () {
-      const minimumVolumeSize = this.minimumVolumeSizeByCloudProfileNameAndRegion({ cloudProfileName: this.cloudProfileName, region: this.region })
-      return parseSize(minimumVolumeSize)
+      const minimumVolumeSize = parseSize(this.minimumVolumeSizeByCloudProfileNameAndRegion({ cloudProfileName: this.cloudProfileName, region: this.region }))
+
+      const machineType = this.selectedMachineType
+      const machineTypeStorageSize = get(machineType, 'storage.size')
+      if (machineTypeStorageSize) {
+        const machineTypeDefaultSize = parseSize(machineTypeStorageSize)
+        if (machineTypeDefaultSize && machineTypeDefaultSize < minimumVolumeSize) {
+          return machineTypeDefaultSize
+        }
+      }
```

Comment:
```suggestion
      const storageSize =  parseSize(get(this.selectedMachineType, 'storage.size'))
      if (storageSize > 0 && storageSize < minimumVolumeSize) {
        return storageSize
      }
```

Reply:
this could potentially lead to a console.error log, or not?

Reply:
will double check